dpnp/backend/extensions/lapack/heevd_batch.cpp

```diff
@@ -0,0 +1,219 @@
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+#include <pybind11/stl.h>
+
+#include "common_helpers.hpp"
+#include "evd_batch_common.hpp"
+#include "heevd_batch.hpp"
+
+// dpctl tensor headers
+#include "utils/type_utils.hpp"
+
+namespace dpnp::extensions::lapack
+{
+namespace mkl_lapack = oneapi::mkl::lapack;
+namespace type_utils = dpctl::tensor::type_utils;
+
+template <typename T, typename RealT>
+static sycl::event heevd_batch_impl(sycl::queue &exec_q,
+                                    const oneapi::mkl::job jobz,
+                                    const oneapi::mkl::uplo upper_lower,
+                                    const std::int64_t batch_size,
+                                    const std::int64_t n,
+                                    char *in_a,
+                                    char *out_w,
+                                    const std::vector<sycl::event> &depends)
+{
+    type_utils::validate_type_for_device<T>(exec_q);
+    type_utils::validate_type_for_device<RealT>(exec_q);
+
+    T *a = reinterpret_cast<T *>(in_a);
+    RealT *w = reinterpret_cast<RealT *>(out_w);
+
+    const std::int64_t a_size = n * n;
+    const std::int64_t w_size = n;
+
+    const std::int64_t lda = std::max<size_t>(1UL, n);
+
+    // Get the number of independent linear streams
+    const std::int64_t n_linear_streams =
+        (batch_size > 16) ? 4 : ((batch_size > 4 ? 2 : 1));
+
+    const std::int64_t scratchpad_size =
+        mkl_lapack::heevd_scratchpad_size<T>(exec_q, jobz, upper_lower, n, lda);
+    T *scratchpad = nullptr;
+
+    // Get padding size to ensure memory allocations are aligned to 256 bytes
+    // for better performance
+    const std::int64_t padding = 256 / sizeof(T);
+
+    if (scratchpad_size > 0) {
+        // Calculate the total scratchpad memory size needed for all linear
+        // streams with proper alignment
+        const size_t alloc_scratch_size =
+            helper::round_up_mult(n_linear_streams * scratchpad_size, padding);
+
+        // Allocate memory for the total scratchpad
+        scratchpad = sycl::malloc_device<T>(alloc_scratch_size, exec_q);
+        if (!scratchpad)
+            throw std::runtime_error("Device allocation for scratchpad failed");
+    }
+    else {
+        throw std::runtime_error(
+            "Invalid scratchpad size: must be greater than zero."
+            "Calculated scratchpad size: " +
+            std::to_string(scratchpad_size));
+    }
```

Comment:
Can we move that to a separate function in `evd_btach_common.hpp`, to have like:
```cpp
T *scratchpad = alloc_scratchpad(scratchpad_size, n_linear_streams, exec_q);
```

Reply:
done